src/App.js

```diff
@@ -38,13 +41,21 @@ export default function App() {
       );
     });
   };
+  const onClose = newTask => {
+    if (newTask) {
+      setTasks(tasks => [...tasks, newTask]);
+    }
+    setModal();
+  };
```

Comment:
basically we'll let the modal component call one function for closing the thing. If the argument is truthy, let's assume it's a new task